tex/main.tex

```diff
@@ -394,60 +340,114 @@ \subsection{Efficient Recursive Least Squares}
-variables introduced in \eqref{eq:sqrtm} results in the following:
-\begin{align}
-    \sqrt{M + W} &= \operatorname{QR_R}\bigg( \begin{bmatrix} \sqrt{M} \\ \sqrt{W} \end{bmatrix} \bigg)
-\end{align}
-Using this, the factorization of the $F^TF$ matrix can be updated using the following
-recursion:
-\begin{align}
-    \sqrt{F_{1:i}^TF_{1:i}} = \operatorname{QR_R}\bigg( \begin{bmatrix} \sqrt{F_{1:i-1}^TF_{1:i-1}} \\ F_i \end{bmatrix} \bigg).
-\end{align}
+where $\operatorname{QR_R}$ returns the upper triangular matrix $R$ from the 
+$\operatorname{QR}$ decomposition. 
+
+We handle regularization of the normal equations, equivalent to adding L2 regularization to 
+the original least squares problem, during the base case of our recursion. If we want to 
+add an L2 regularization with weight $\rho$, we calculate $U_1$ as:
+
+\begin{equation}
+    U_1 =  \operatorname{QR_R}\bigg( 
+        \begin{bmatrix} \sqrt{F_1} \\ \sqrt{\rho} I \end{bmatrix}.
+    \bigg),
+\end{equation}
+
 The final algorithm for solving a least squares problem in a recursive-batch fashion is
 described in algorithm \ref{alg:rlsqr}. This algorithm can be modified to handle L2
 regularized least squares problems by simply replacing line 2 of algorithm \ref{alg:rlsqr}
 with $U\leftarrow \operatorname{QR_R}(\operatorname{vcat}(F_1,\sqrt{\rho}I))$, where $\rho$
 is the regularizer. 
  \begin{algorithm} 
-	\begin{algorithmic}[1]
-		\caption{Recursive Batch Least Squares with QR}\label{alg:rlsqr}
-		\State \textbf{input} $F,\,d$  \Comment{problem data}
-		\State $U \leftarrow \operatorname{QR_R}(F_1)$ \Comment{form initial upper-triangular square-root}
-		\State $b \leftarrow F_1^Td_1$ \Comment{form initial right hand side vector}
-		\For{$i = 2:N$}
+    \begin{algorithmic}[1]
+        \caption{Recursive Batch Least Squares with QR}\label{alg:rlsqr}
+        \State \textbf{input} $F,\,d$  \Comment{problem data}
+        \State $U \leftarrow \operatorname{QR_R}(\operatorname{vcat}(F_1,\sqrt{\rho}I))$ \Comment{form initial upper-triangular square-root}
+        \State $b \leftarrow F_1^Td_1$ \Comment{form initial right hand side vector}
+        \For{$i = 2:N$}
 % 		\State $U \leftarrow \operatorname{QR_R}([U^T \, A_i^T]^T) $ 
         \State $U \leftarrow \operatorname{QR_R}(\operatorname{vcat}(U,F_i)) $ \Comment{update square-root with new batch}
-		\State $b \leftarrow b + F_i^Td_i$ \Comment{update right hand side with new batch}
-		\EndFor
-		\State \textbf{ouput} \,$x \leftarrow U^{-1}U^{-T}b$ \Comment{forward and backwards substitution to solve for $x$}
-	\end{algorithmic}
+        \State $b \leftarrow b + F_i^Td_i$ \Comment{update right hand side with new batch}
+        \EndFor
+        \State \textbf{ouput} \,$x \leftarrow U^{-1}U^{-T}b$ \Comment{forward and backwards substitution to solve for $x$}
+    \end{algorithmic}
 \end{algorithm}
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

Comment:
Since we have a bit of repeated stuff between the two systems, I'm starting to think that organizing by topic, such as "Sample Efficiency", "Generalization", "Training Robustness" would be more effective than grouping by system. By doing some scaling / normalization tricks, we could plot the quadrotor and cartpole results together in some cases, which would save room and be easier to parse.

Reply:
I'm down for this - definitely agree on it being better for trimming the fat

Reply:
I like this too